src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,958 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+    data['hybridConnectionConfig']['hybridConnectionName'] = response.hybrid_connection_config.hybrid_connection_name
+    data['hybridConnectionConfig']['token'] = response.hybrid_connection_config.token
+    data['hybridConnectionConfig']['expirationTime'] = response.hybrid_connection_config.expiration_time
+    return data
+
+
+def client_side_proxy_main(cmd,
+                           client,
+                           resource_group_name,
+                           cluster_name,
+                           flag,
+                           args,
+                           client_proxy_port,
+                           api_server_port,
+                           operating_system,
+                           creds,
+                           user_type,
+                           debug_mode,
+                           token=None,
+                           path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                           context_name=None,
+                           clientproxy_process=None):
+    expiry, clientproxy_process = client_side_proxy(cmd, client, resource_group_name, cluster_name, 0, args, client_proxy_port, api_server_port, operating_system, creds, user_type, debug_mode, token=token, path=path, context_name=context_name, clientproxy_process=None)
+    next_refresh_time = expiry - consts.CSP_REFRESH_TIME
+    
+    while(True):
+        time.sleep(60)
+        if(check_if_csp_is_running(clientproxy_process)):
+            if time.time() >= next_refresh_time:
+                expiry, clientproxy_process = client_side_proxy(cmd, client, resource_group_name, cluster_name, 1, args, client_proxy_port, api_server_port, operating_system, creds, user_type, debug_mode, token=token, path=path, context_name=context_name, clientproxy_process=clientproxy_process)
+                next_refresh_time = expiry - consts.CSP_REFRESH_TIME
+        else:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception='Process closed externally.', fault_type=consts.Proxy_Closed_Externally_Fault_Type,
+                                    summary='Process closed externally.')
+            raise CLIError('Proxy closed externally.')
+
+
+def client_side_proxy(cmd,
+                      client,
+                      resource_group_name,
+                      cluster_name,
+                      flag,
+                      args,
+                      client_proxy_port,
+                      api_server_port,
+                      operating_system,
+                      creds,
+                      user_type,
+                      debug_mode,
+                      token=None,
+                      path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                      context_name=None,
+                      clientproxy_process=None):
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if token is not None:
```

Comment:
can we move this telemetry check to client_side_proxy_wrapper ?. Otherwise we will be sending telemetry refresh every 3 hours ?

Reply:
removed the extension event altogether as its not needed anymore.